src/js/subtraction/components/SubtractionFileSelector.tsx

```diff
@@ -0,0 +1,115 @@
+import { flatMap } from "lodash-es";
+import React, { useEffect } from "react";
+import { InfiniteData } from "react-query";
+import { FetchNextPageOptions, InfiniteQueryObserverResult } from "react-query/types/core/types";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+import { BoxGroup, InputError, LoadingPlaceholder, NoneFoundBox } from "../../base";
+import { ScrollList } from "../../base/ScrollList";
+import { useInfiniteFindFiles } from "../../files/querys";
+import { FileResponse, FileType } from "../../files/types";
+import { SubtractionFileItem } from "./SubtractionFileItem";
+
+const SubtractionFileSelectorError = styled(InputError)`
+    margin-bottom: 5px;
+`;
+
+const SubtractionFileSelectorList = styled(BoxGroup)`
+    max-height: 400px;
+    overflow-y: auto;
+`;
+
+function getAllSubtractions(fetchNextPage, hasNextPage, data) {
+    if (hasNextPage) {
+        void fetchNextPage();
+    }
+    return data.pages.flatMap(page => page.items);
+}
+
+type SubtractionFileSelectorProps = {
+    /** The subtraction files */
+    files: InfiniteData<FileResponse>;
+    /** The number of subtraction files */
+    foundCount: number;
+    /** The selected file id */
+    selected: string;
+    /** A callback function to handle file selection */
+    onClick: (selected: string) => void;
+    /** Errors occurred on sample creation */
+    error: string;
+    /** Fetches the next page of data */
+    fetchNextPage: (options?: FetchNextPageOptions) => Promise<InfiniteQueryObserverResult>;
+    /** Whether the data is loading */
+    isLoading: boolean;
+    /** Whether the next page is being fetched */
+    isFetchingNextPage: boolean;
+};
+
+/**
+ * A list of subtraction files
+ */
+export function SubtractionFileSelector({
+    files,
+    foundCount,
+    selected,
+    onClick,
+    error,
+    fetchNextPage,
+    isLoading,
+    isFetchingNextPage,
+}: SubtractionFileSelectorProps) {
+    const {
+        data: subtractions,
+        isLoading: isLoadingSubtractions,
+        fetchNextPage: fetchNextSubtractionsPage,
+        hasNextPage,
+    } = useInfiniteFindFiles(FileType.subtraction, 25);
+
+    useEffect(() => {
+        if (!isLoadingSubtractions && selected) {
+            const allSubtractions = getAllSubtractions(fetchNextSubtractionsPage, hasNextPage, subtractions);
```

Comment:
Naming and signature of this function is misleading imo. Since the underlying really just returns the data it has already been passed.

Since you'll need something similar in the samples view it might be worth considering a custom hook along the lines of `useValidateFiles` that is capable of generically handling out of date fiel selections. Not required for merge though